TeamCode/src/main/java/org/firstinspires/ftc/teamcode/OpModes/Auto/Basic/Sampling.java

```diff
@@ -0,0 +1,119 @@
+
+import org.firstinspires.ftc.robotcore.external.tfod.Recognition;
+import org.firstinspires.ftc.teamcode.Components.IntakeLift;
+import org.firstinspires.ftc.teamcode.OpModes.Auto.Utils.AutoMode;
+
+import java.util.List;
+
+/**
+ * AUTO MODE: Sampling
+ *
+ * POINT WORTH: 15 pts
+ *
+ * SETUP: Front facing the sample
+ *
+ * SUMMARY: The auto uses TF to detect from the sample,
+ *  the position of the gold mineral and turns towards the gold minerals position.
+ *  The robot will then drive forward.
+ */
+
+@Autonomous(name = "Sampling", group = "Official")
+public class Sampling extends AutoMode {
+
+    //Auto specific
+    private boolean sampleComplete = false;
+    private enum Turn{
+        LEFT(-90), RIGHT(90);
+
+        private final int degrees;
+        Turn(int degrees) {
+            this.degrees = degrees;
+        }
+        public int getDegrees() {
+            return degrees;
+        }
+    }
+    Turn turn;
+
+    @Override
+    public void init() {
+        super.init();
+    }
+
+    @Override
+    public void loop() {
+        switch(states){
+            case START:
+                telemetry.addData("Current State","Start");
+                intakeLift.setAutoPos(IntakeLift.AutoPos.UP);
+                nextState(true, States.VISION);
+                break;
+
+            case VISION:
+                telemetry.addData("Current State", "Vision");
+                if (tfod != null) {
+                    tfod.activate();
+                    List<Recognition> updatedRecognitions = tfod.getUpdatedRecognitions();
+                    if (updatedRecognitions != null) {
+                        telemetry.addData("# Object Detected", updatedRecognitions.size());
+                        if (updatedRecognitions.size() == 3) {
+                            int goldMineralX = -1;
+                            int silverMineral1X = -1;
+                            int silverMineral2X = -1;
+                            for (Recognition recognition : updatedRecognitions) {
+                                if (recognition.getLabel().equals(LABEL_GOLD_MINERAL)) {
+                                    goldMineralX = (int) recognition.getLeft();
+                                } else if (silverMineral1X == -1) {
+                                    silverMineral1X = (int) recognition.getLeft();
+                                } else {
+                                    silverMineral2X = (int) recognition.getLeft();
+                                }
+                            }
+                            if (goldMineralX != -1 && silverMineral1X != -1 && silverMineral2X != -1) {
+                                if (goldMineralX < silverMineral1X && goldMineralX < silverMineral2X) {
+                                    telemetry.addData("Gold Mineral Position", "Right");
+                                    turn = Turn.RIGHT;
+                                    nextState(true, States.TURN);
```

Comment:
Why is this 
```java
nextState(true, States.TURN);
```
and the other ones are
```java
states = States.TURN;
```
and 
```java
states = States.DRIVE;
```

I think they should all be consistent and use `nextState()`

Reply:
@GarrettBurroughs I forgot to finish them earlier, my b